8.frontend-react/crm-frontend/src/screens/AllProjects.jsx

```diff
@@ -0,0 +1,176 @@
+   
+   const onRemoveItem = (value) => {
+     setItemToDelete(value);
+     openDeleteProjectWindow();
+   }
+   
+   const removeItem = () => {
+    // authApi.deleteUser(itemToDelete);
+    // console.log("item to delete", itemToDelete);
+    // let newData = dataRef.current.filter((item)=>{
+    //   console.log(itemToDelete.project_id,"is equal?", item.user_id);
+    //   return itemToDelete.user_id != item.user_id;
+    // })
+    // closeDeleteProjectWindow();
+    // setData(newData);
+   }
+   
+   useEffect(()=>{
+     (async () => {
+      const result = await getProjectsList();
+      setData(result);
+     })();
+   }, [props.mine])
+    
+  
+   const columns = React.useMemo(
+    () => [
+      {
+        Header: 'Type',
+        accessor: 'item_type', // accessor is the "key" in the data
+
+      },
+      {
+        Header: 'Client',
+        accessor: 'client_name',
+      },
+      {
+        Header: 'Date',
+        accessor: 'created',
+        Cell: ({value}) => {
+            return parseDate(value);
+        }
+      },
+      {
+        Header: 'Deadline',
+        accessor: 'deadline',
+        Cell: ({value}) => {
+            return parseDate(value);
+        }
+      },
+      {
+        Header: 'Status',
+        accessor: 'project_status',
+        Cell: ({value}) => {
+            console.log(props.mine);
+            return mineRef.current ? "in progress" : value;
+        }
+        // Cell: ({value})  => value == 'active' ? <FontAwesomeIcon className='status-icon' icon={faCheck} size='xs'/> : value
+      },
+      {
+        Header: 'Action',
+        // accessor: 'delete',
+        Cell: (value)=> (
+          <div>
+            <span style={{cursor:'pointer'}}
+                onClick={() => {
+                  // console.log(value);
+                    onRemoveItem(value.cell.row.original);
+                  }}>
+                  <FontAwesomeIcon className='trash-icon' icon={faTrash} size='sm'/>
+          </span> 
+          </div>
+          
+        )
+      },
+
+    ],
+    []
+  )
+
+
+    const openDeleteProjectWindow = ()=>{
+      setIsDeleteModalOpen(true);
+    };
+
+    const closeDeleteProjectWindow = ()=>{
+        setIsDeleteModalOpen(false);
+    };
+
+    
+
+    return (
+        <div>
+            <Header/>
+            <div className='crm-page'>
+            <PageTitle className='page-title' title={props.mine ? 'My Projects' : 'All Projects'} description='Manage your projects.'/>
+            {!props.mine && <div className='add-user-box'>
+            <CrmButton content='Add Project' buttonClass='main-button' icon='plus' isLoading={false} callback={()=> {}}/>
+            </div>}
+            <Table columns={columns} data={data}/>
+            <Modal isOpen={isDeleteModalOpen} ariaHideApp={false} contentLabel='Remove Project' onRequestClose={closeDeleteProjectWindow}  overlayClassName="Overlay" className='modal'>
+                <h2>Are you sure you want delete this item?</h2>
```

Comment:
you gonna have this "are you sure you want to delete modal in a lot of place. 
think about refactoring it to it's own component that can be used anywhere
the component will get a title , and action on delete, and is open, an will be able to handle this modal completely :)